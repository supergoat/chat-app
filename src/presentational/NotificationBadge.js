// @flow
import React, { Component } from 'react';
import {
  Text,
  View
} from 'react-native';
import CAColors from '../common/CAColors';
import CAStyleSheet from '../common/CAStyleSheet';

/* Constants
============================================================================= */
const BADGE_HEIGHT = 24;
const BADGE_MIN_WIDTH = BADGE_HEIGHT;
const BADGE_BORDER_RADIUS = BADGE_HEIGHT / 2;

/** ============================================================================
<NotificationBadge
  noOfNotifications={number}
  [style]={object}
/>
--------------------------------------------------------------------------------
Displays the number of notifications to the user
@param { string } noOfNotifications - The number of notifications
@param { object } [style] - custom style
============================================================================= */
type Props = {
  noOfNotifications: number,
  style?: { [name: string]: any }
};
class NotificationBadge extends Component<Props> {
  static __cards__: Function;

  render() {
    const { noOfNotifications, style } = this.props;
    return (
      noOfNotifications > 0 && (
        <View style={[styles.badge, style]}>
          <Text style={styles.badgeNumber}>
            {noOfNotifications < 100 ? noOfNotifications : '+99'}
          </Text>
        </View>
      )
    );
  }
}

/* StyleSheet
============================================================================= */
const styles = CAStyleSheet.create({
  badge: {
    backgroundColor: CAColors.apple,
    alignItems: 'center',
    justifyContent: 'center',
    minWidth: BADGE_MIN_WIDTH,
    height: BADGE_HEIGHT,
    borderRadius: BADGE_BORDER_RADIUS,
    alignSelf: 'flex-start' // prevent expanding to fill space
  },
  badgeNumber: {
    color: CAColors.white,
    fontSize: 10,
    paddingRight: 5,
    paddingLeft: 5,
    fontWeight: 'bold'
  }
});

/* Playground Cards - the different states of NotificationBadge
============================================================================= */
NotificationBadge.__cards__ = define => {
  define('Number of notifications less than 1', () => {
    return <NotificationBadge noOfNotifications={0} />;
  });

  define('Number of notifications less than 100', () => {
    return <NotificationBadge noOfNotifications={10} />;
  });

  define('Number of notifications over 99', () => {
    return <NotificationBadge noOfNotifications={100} />;
  });
};

/* Export
============================================================================= */
export default NotificationBadge;
