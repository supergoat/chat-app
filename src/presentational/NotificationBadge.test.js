// @flow
import React from 'react';
import NotificationBadge from './NotificationBadge';
import renderer from 'react-test-renderer';

describe('NotificationBadge', () => {
  it("does't display notifications when noOfNotifications is 0", () => {
    const tree = renderer
      .create(<NotificationBadge noOfNotifications={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('displays notifications in a circle when less than 100', () => {
    const tree = renderer
      .create(<NotificationBadge noOfNotifications={10} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('displays +99 in a rounded rectangle when over 99', () => {
    const tree = renderer
      .create(<NotificationBadge noOfNotifications={100} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
