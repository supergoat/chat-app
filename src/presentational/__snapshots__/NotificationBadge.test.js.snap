// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBadge displays +99 in a rounded rectangle when over 99 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#32CD32",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 10,
        "fontWeight": "bold",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    +99
  </Text>
</View>
`;

exports[`NotificationBadge displays notifications in a circle when less than 100 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "backgroundColor": "#32CD32",
        "borderRadius": 12,
        "height": 24,
        "justifyContent": "center",
        "minWidth": 24,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 10,
        "fontWeight": "bold",
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    10
  </Text>
</View>
`;

exports[`NotificationBadge does't display notifications when noOfNotifications is 0 1`] = `null`;
